### ISSUE TITLE
native ubuntu 2004 machines?

### ISSUE THREAD

**Issue by mhdawson** [nodejs/build#2895]
I had to drop 1804 out of the node-addon ci due to the kernel levels of the release binaries for main/18.

I noticed we don't have any 2004 that I can add in instead.  Should we have tom ubuntu 2004 machines?

**Comment by richardlau** [comment]
Yes, we should. I'd like to retire the Ubuntu 1404, and ideally Ubuntu 1604 test machines and repurpose for Ubuntu 20.04. @mhdawson can you check if the node-addon CI is using any of those?

**Comment by mhdawson** [comment]
It looks like we still use ubuntu1604 to test 14 and 12. I could remove that but then we would not have any older kernel testing on x64.  We do test on centos7 for ppcle, s390 but I don't see any centos7 x64 machines that I could replace the ubuntu ones. It's kind of interesting that we release on centos7 for x64 on older relases but we don't seem to build/test on it.

1404 is not used in node-addon-api testing.

**Comment by sxa** [comment]
Bear in mind that since the next Ubuntu LTS (22.04) is due out in a week's time. On that basis if we're going to migrate some systems up I suggest we get a mix of 22.04 and 20.04 systems.

> It's kind of interesting that we release on centos7 for x64 on older relases but we don't seem to build/test on it.

Yeah that seems like quite an obvious omission if it's correct. CentOS7 is an earlier kernel than Ubuntu 16.04 and it's still in support, so sorting that out seems like an obvious thing to do.

**Comment by richardlau** [comment]
> Bear in mind that since the next Ubuntu LTS (22.04 "Jammy Jellyfish") is due out in a week's time. On that basis if we're going to migrate some systems up I suggest we get a mix of 22.04 and 20.04 systems.
> 

Agreed, although I assume the `.04` means it's due out next month 🙂. 

> > It's kind of interesting that we release on centos7 for x64 on older relases but we don't seem to build/test on it.
> 
> Yeah that seems like quite an obvious omission if it's correct. CentOS7 is an earlier kernel than Ubuntu 16.04 and it's still in support, so sorting that out seems like an obvious thing to do.

We have two CentOS 7 x64 test instances:
- [test-softlayer-centos7-x64-1](https://ci.nodejs.org/computer/test-softlayer-centos7-x64-1)
- [test-rackspace-centos7-x64-1](https://ci.nodejs.org/computer/test-rackspace-centos7-x64-1)

For some reason (perhaps they predate the `select-compiler.sh` script) [node-test-commit-linux](https://ci.nodejs.org/job/node-test-commit-linux/) uses labels `centos7-64-gcc6` and `centos7-64-gcc8` (and `centos7-64-gcc48` but I'm fairly certain nothing should be using that) although the Jenkins nodes are also labelled `centos7-64`.

**Comment by richardlau** [comment]
I think in the short term repurposing the two ubuntu1404 test instances to Ubuntu 20.04 would be a good initial step. I'll take another sweep through the CI job configs to make sure nothing is referencing them.

Maybe Node.js 12 going End-of-Life would be a good point to replace Ubuntu 16.04 with 22.04.

**Comment by mhdawson** [comment]
I can try adding in those 2 centos tags to see what runs.

**Comment by sxa** [comment]
> Agreed, although I assume the `.04` means it's due out next month slightly_smiling_face.

Yep - I'm getting ahead of myself - apologies. I was thinking we were already in Aprill ...
